ospf6d/ospf6_interface.c

```diff
@@ -154,6 +154,15 @@ static uint32_t ospf6_interface_get_cost(struct ospf6_interface *oi)
 			cost = 1;
 	}
 
+	/* If the interface type is point-to-multipoint or the interface
+	 * is in the state Loopback, the global scope IPv6 addresses
+	 * associated with the interface (if any) are copied into the
+	 * intra-area-prefix-LSA with the PrefixOptions LA-bit set, the
+	 * PrefixLength set to 128, and the metric set to 0.
+	 */
+	if (if_is_loopback(oi->interface))
```

Comment:
Isn't it better to move this code block at the beginning of the function and return instead of doing all the checks and assignment and then overwriting it with 0.

Reply:
Absolutely agree, fixed, thanks.